Grbl_Esp32/Custom/oled_basic.cpp

```diff
@@ -136,74 +116,127 @@ String getStateText() {
-        display.drawString(0, row3, radio_info);
-        //}
+        radio_name = "STA: " + WiFi.SSID();
+        radio_addr = WiFi.localIP().toString();
     } else if ((WiFi.getMode() == WIFI_MODE_AP) || (WiFi.getMode() == WIFI_MODE_APSTA)) {
-        radio_info = String("AP SSID: ") + WebUI::wifi_ap_ssid->get();
-
-        display.drawString(0, row1, radio_info);
-
-        radio_info = "IP: " + WiFi.softAPIP().toString();
-        display.drawString(0, row2, radio_info);
+        radio_name = String("AP:") + WebUI::wifi_ap_ssid->get();
+        radio_addr = WiFi.softAPIP().toString();
     }
 #endif
 
 #ifdef WIFI_OR_BLUETOOTH
     if (WebUI::wifi_radio_mode->get() == ESP_RADIO_OFF) {
-        display.drawString(0, row1, "Radio Mode: None");
+        radio_name = "Radio Mode: None";
     }
 #else
-    display.drawString(0, row1, "Wifi and Bluetooth Disabled");
+    radio_name = "Radio Mode:Disabled";
 #endif
+
+    display.setTextAlignment(TEXT_ALIGN_LEFT);
+    display.setFont(ArialMT_Plain_10);
+
+    if (sys.state == State::Alarm) {  // print below Alarm:
+        display.drawString(0, 18, radio_name);
+        display.drawString(0, 30, radio_addr);
+
+    } else {  // print next to status
+        if (WebUI::wifi_radio_mode->get() == ESP_BT) {
+            display.drawString(55, 2, radio_name);
+        } else {
+            display.drawString(55, 2, radio_addr);
+        }
+    }
+}
+// Here changes begin  Here changes begin Here changes begin Here changes begin Here changes begin
+
+void draw_checkbox(int16_t x, int16_t y, int16_t width, int16_t height, bool checked) {
+    if (checked)
+        display.fillRect(x, y, width, height);  // If log.0
+    else
+        display.drawRect(x, y, width, height);  // If log.1
 }
 
 void displayDRO() {
-    char axisVal[20];
+    uint8_t oled_y_pos;
+    float   print_position[MAX_N_AXIS];
+    float   wco[MAX_N_AXIS];
 
-    display.setFont(ArialMT_Plain_16);
     display.setTextAlignment(TEXT_ALIGN_LEFT);
-    display.drawString(0, 13, String('X') + ":");
-    display.drawString(0, 30, "Y:");
-    display.drawString(0, 47, "Z:");
+    display.setFont(ArialMT_Plain_10);
+
+    char axisVal[20];
 
-    display.setTextAlignment(TEXT_ALIGN_RIGHT);
-    snprintf(axisVal, 20 - 1, "%.3f", getPosition(X_AXIS, true));
-    display.drawString(100, 13, axisVal);
+    display.drawString(80, 14, "L");  // Limit switch
 
-    snprintf(axisVal, 20 - 1, "%.3f", getPosition(Y_AXIS, true));
-    display.drawString(100, 30, axisVal);
+    auto        n_axis         = number_axis->get();
+    AxisMask    lim_pin_state  = limits_get_state();
+    ControlPins ctrl_pin_state = system_control_get_state();
+    bool        prb_pin_state  = probe_get_state();
 
-    snprintf(axisVal, 20 - 1, "%.3f", getPosition(Z_AXIS, true));
-    display.drawString(100, 47, axisVal);
+    report_calc_status_position(print_position, wco, bit_isfalse(status_mask->get(), RtStatus::Position));
```

Comment:
I'm wondering if it is a good idea to make the OLED display format contingent on the serial report format.  It might be better to have simpler "calc_mpos(print_position)" and "calc_wpos(print_position, wco)" functions.  The messy logic that status reports use, involving both the status mask and that funky counter, could then be isolated in the status reporting code, i.e. 
```
  if (bit_isfalse(status_mask->get(), RtStatus::Position) || (sys.report_wco_counter == 0)) {
    calc_wpos(print_position, wco);
  } else {
    calc_mpos(print_position);
  }
```
It seems to me that the status report format stuff is an interface contract with the sender.  Using that same "knob" to control the OLED display seems like a recipe for testing problems and strange OLED behavior when a given sender changes the report format according to its needs.